In [1]:
import numpy as np
from mostlyai import MostlyAI

In [2]:
# initialize client
mostly = MostlyAI(api_key='mostly-adcd77a4ca7523519dc19b4ec9c9a7f6e0b2e37e16599b4703b03944dc824ed2', base_url='https://app.mostly.ai')

In [3]:
# fetch configuration via API
g = mostly.generators.get('6a19773e-9bd7-402a-aa65-a75e1e688c26')
config = g.config()
config

{'name': 'cleaned_ipums_data',
 'description': None,
 'tables': [{'name': 'cleaned_ipums_data',
   'sourceConnectorId': 'ac9bdefa-05d9-4c98-bc55-f1740854a025',
   'location': 'mostly-production-mostly-s3-bucket-mostly-app/connectors/ac9bdefa-05d9-4c98-bc55-f1740854a025/cleaned_ipums_data/cleaned_ipums_data.csv',
   'data': None,
   'modelConfiguration': {'maxSampleSize': 197756,
    'batchSize': None,
    'modelSize': 'M',
    'maxTrainingTime': 10,
    'maxEpochs': 100,
    'maxSequenceWindow': 100,
    'enableFlexibleGeneration': False,
    'valueProtection': False,
    'rareCategoryReplacementMethod': 'CONSTANT',
    'compute': 'd5f093a6-208f-4b79-92c6-6d50d02d8c8c'},
   'textModelConfiguration': None,
   'primaryKey': None,
   'foreignKeys': None,
   'columns': [{'name': 'incwage', 'modelEncodingType': 'NUMERIC_BINNED'},
    {'name': 'years_of_educ', 'modelEncodingType': 'NUMERIC_DISCRETE'},
    {'name': 'potential_experience', 'modelEncodingType': 'NUMERIC_BINNED'},
    {'name': '

In [ ]:
# probe the generator for some samples
mostly.probe('6a19773e-9bd7-402a-aa65-a75e1e688c26', size=10)

We need 20 synthetic data sets. Original data set size was 197,756, so we need $197,756 \times 20 = 3,955,120$ samples.

Due to limits on the free version, we will break it into four requests. This will be done manually over 4 days.

In [4]:
# use generator to create a synthetic dataset
sd = mostly.generate('6a19773e-9bd7-402a-aa65-a75e1e688c26', size=988_780)
sd.data()

Created synthetic dataset ]8;id=902581;https://app.mostly.ai/d/synthetic-datasets/efba1a97-9233-4258-b74e-4d149cb7d5af\efba1a97-9233-4258-b74e-4d149cb7d5af]8;;\ with generator ]8;id=794938;https://app.mostly.ai/d/generators/6a19773e-9bd7-402a-aa65-a75e1e688c26\6a19773e-9bd7-402a-aa65-a75e1e688c26]8;;\

Started synthetic dataset generation

Output()

🎉 Your synthetic dataset is ready! Use it to consume the generated data. Share it so others can do the same.

incwage  years_of_educ  potential_experience  non_white  sex
0     -0.353971      -0.429516             -0.593985          1    1
1     -0.658441      -0.044869              0.287642          1    0
2     -0.100254       0.339777              0.290098          0    0
3     -0.915482      -0.429516              0.467786          0    1
4     -0.952355      -0.429516             -1.408404          0    1
...         ...            ...                   ...        ...  ...
98873 -0.781871      -0.429516             -1.328339          0    0
98874 -0.259111      -0.429516              0.696675          0    1
98875 -0.657376      -0.429516              0.717362          0    0
98876 -0.992202      -0.429516              -1.51103          0    0
98877   0.22899      -0.429516             -1.088247          0    0

[988780 rows x 5 columns]

Split the generated synthetic data into 20 data sets.

In [5]:
split_sd = np.array_split(sd.data(), 5)

C:\Users\Cameron\anaconda3\Lib\site-packages\numpy\core\fromnumeric.py:59: FutureWarning: 'DataFrame.swapaxes' is deprecated and will be removed in a future version. Please use 'DataFrame.transpose' instead.
  return bound(*args, **kwds)


In [6]:
split_sd[0]

incwage  years_of_educ  potential_experience  non_white  sex
0     -0.353971      -0.429516             -0.593985          1    1
1     -0.658441      -0.044869              0.287642          1    0
2     -0.100254       0.339777              0.290098          0    0
3     -0.915482      -0.429516              0.467786          0    1
4     -0.952355      -0.429516             -1.408404          0    1
...         ...            ...                   ...        ...  ...
98873 -0.001395      -0.044869              0.460547          0    1
98874 -0.395288      -1.583456             -0.118807          1    0
98875 -0.959724      -0.044869              -1.51103          0    1
98876 -0.024554      -0.429516             -1.341637          0    1
98877 -0.786854       -1.19881              0.055851          1    0

[197756 rows x 5 columns]

# MUST CHANGE INDEX WHEN SAVING ADDITIONAL DATASETS SO YOU DON'T OVERWRITE EXISTING

In [7]:
for i, sX in enumerate(split_sd):
    sX.to_csv("../Data/IPUMS/Synthetic Datasets/mostly_" + str(i) + ".csv", index=False)